coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,658 @@
+from contextlib import contextmanager
+import inspect
+from itertools import compress
+import re
+import shutil
+from subprocess import check_call, CalledProcessError, DEVNULL
+from types import MappingProxyType
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.misc.ContextManagers import make_temp
+from coalib.misc.Decorators import assert_right_type, enforce_signature
+from coalib.misc.Future import partialmethod
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+def _prepare_options(options):
+    """
+    Prepares options for ``Linter`` for a given options dict in-place.
+
+    :param options:
+        The options dict that contains user/developer inputs.
+    """
+    allowed_options = {"executable",
+                       "output_format",
+                       "use_stdin",
+                       "use_stdout",
+                       "use_stderr",
+                       "config_suffix",
+                       "prerequisite_check_command"}
+
+    if not options["use_stdout"] and not options["use_stderr"]:
+        raise ValueError("No output streams provided at all.")
+
+    if options["output_format"] == "corrected":
+        if (
+                "diff_severity" in options and
+                options["diff_severity"] not in RESULT_SEVERITY.reverse):
+            raise TypeError("Invalid value for `diff_severity`: " +
+                            repr(options["diff_severity"]))
+
+        if "diff_message" in options:
+            assert_right_type(options["diff_message"], str, "diff_message")
+
+        allowed_options |= {"diff_severity", "diff_message"}
+    elif options["output_format"] == "regex":
+        if "output_regex" not in options:
+            raise ValueError("`output_regex` needed when specified "
```

Comment:
@AbdealiJK file an issue? That's a low one and I want this merged ASAP, it's not a big issue is it? Just a simplification.